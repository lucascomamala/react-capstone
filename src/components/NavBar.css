* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  list-style-type: none;
}

a {
  text-decoration: none;
}

.home-link {
  font-weight: 900;
}

.home-link:hover {
  color: rgb(103, 24, 24);
}

.navigation-bar {
  background-color: rgba(61, 61, 61, 0.7);
  background-image: url(/public/media/grid-noise.png);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
}

.navigation-ul {
  display: flex;
  gap: 30px;
}

.navigation-ul a {
  text-decoration: none;
  color: black;
}

.nav-logo {
  height: 100px;
}

.home-navlink {
  font-size: 1.5rem;
  font-family: Papyrus, Times, serif;
  color: black;
}

@media all and (min-width: 768px) {
  .navigation-bar {
    background-color: rgb(255, 255, 255, 0.7);
    background-image: linear-gradient(90deg, rgba(0, 0, 0, 0.8), transparent, rgba(0, 0, 0, 0.8)), url(/public/media/grid-noise.png);
    position: fixed;
    margin: auto;
    z-index: 100;
    width: 700px;
    left: 50%;
    top: 10px;
    margin-left: -350px;
    padding: 5px 30px;
    border-radius: 30px;
    box-shadow: 0 0 15px black;
  }
}
