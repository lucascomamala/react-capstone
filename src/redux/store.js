import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import champions from './HomeSlice';

const store = configureStore({
  reducer: {
    champions,
  },
  middleware: [thunk, logger],
});

export default store;
