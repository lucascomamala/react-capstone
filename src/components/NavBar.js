/* eslint-disable */
import { Link, useLocation } from 'react-router-dom';
import './NavBar.css';

const Navbar = () => {
  const location = useLocation();
  return (
    <nav className="navigation-bar">
      <Link to="/" className="home-link">
        <span className="home-navlink">{location.pathname === '/' ? 'HOME' : 'GO BACK'}</span>
      </Link>
      <Link to="/" className="home-link">
        <img className="nav-logo" src="/media/logo.png" alt="league-logo" />
      </Link>
    </nav>
  );
};

export default Navbar;
