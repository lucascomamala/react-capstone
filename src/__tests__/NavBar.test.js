/* eslint-disable import/extensions */
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import NavBar from '../components/NavBar';
import store from '../redux/store';

describe('Navbar component', () => {
  test('Renders Navbar component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <NavBar />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
