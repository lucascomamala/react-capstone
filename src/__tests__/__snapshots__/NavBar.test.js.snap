// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navigation-bar"
      >
        <a
          class="home-link"
          href="/"
        >
          <span
            class="home-navlink"
          >
            HOME
          </span>
        </a>
        <a
          class="home-link"
          href="/"
        >
          <img
            alt="league-logo"
            class="nav-logo"
            src="/media/logo.png"
          />
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navigation-bar"
    >
      <a
        class="home-link"
        href="/"
      >
        <span
          class="home-navlink"
        >
          HOME
        </span>
      </a>
      <a
        class="home-link"
        href="/"
      >
        <img
          alt="league-logo"
          class="nav-logo"
          src="/media/logo.png"
        />
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
